app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java

```diff
@@ -417,9 +431,32 @@ public void handleResult(@NonNull final ChannelInfo result) {
 
         playlistCtrl.setVisibility(View.VISIBLE);
 
-        if (!result.getErrors().isEmpty()) {
-            showSnackBarError(result.getErrors(), UserAction.REQUESTED_CHANNEL,
-                    NewPipe.getNameOfService(result.getServiceId()), result.getUrl(), 0);
+        List<Throwable> errors = result.getErrors();
+        if (!errors.isEmpty()) {
+
+            // handling ContentNotSupportedException not to show the error but an appropriate string
+            // so that crashes won't be sent uselessly and the user will understand what happened
+            Iterator<Throwable> it = errors.iterator();
+            Throwable contentNotSupported = null;
+            while (it.hasNext()) {
+                Throwable throwable = it.next();
+                if (throwable instanceof ContentNotSupportedException) {
+                    showContentNotSupported();
+                    contentNotSupported = throwable;
+                    it.remove();
+                }
+            }
+
+            if (!errors.isEmpty()) {
+                showSnackBarError(result.getErrors(), UserAction.REQUESTED_CHANNEL,
+                        NewPipe.getNameOfService(result.getServiceId()), result.getUrl(), 0);
+            }
+
+            if (contentNotSupported != null) {
```

Comment:
Instead of adding the error at the end, it might be good to keep the original order. We should create a copy the error list.